src/librustc_borrowck/borrowck/check_loans.rs

```diff
@@ -395,18 +395,40 @@ impl<'a, 'tcx> CheckLoanCtxt<'a, 'tcx> {
         assert!(self.bccx.region_scope_tree.scopes_intersect(old_loan.kill_scope,
                                                        new_loan.kill_scope));
 
-        self.report_error_if_loan_conflicts_with_restriction(
-            old_loan, new_loan, old_loan, new_loan) &&
-        self.report_error_if_loan_conflicts_with_restriction(
-            new_loan, old_loan, old_loan, new_loan)
+        let err_old_new = match self.report_error_if_loan_conflicts_with_restriction(
```

Comment:
These `match`es can be changed to call [`Result::err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.err) instead.